packages/storage/amplify_storage_s3/lib/src/S3PrefixResolver/prefix_handler.dart

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_core/amplify_core.dart';
+
+import '../types.dart';
+
+class PrefixHandler {
+  static Future<dynamic> handlePrefix(Map<String, dynamic> arguments,
```

Comment:
Classes with just static methods is generally considered an anti-pattern. There's even a lint for it: https://dart.dev/tools/linter-rules#avoid_classes_with_only_static_members. Is there a reason this shouldn't be defined on the method channel?

Reply:
The intention was to avoid 'polluting' the method channel class and to therefore separate out the prefix handler logic to another class. 

It's really not necessary since we're going to deprecate this code soon anyway so I can merge this back in.